### PR TITLE
Sync with template

### PR DESCRIPTION
**Pull request by ahuang11**
Cruft update and add landing page
image

### Checklist

- [ ] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-email/issues/new/choose) first.
- [ ] Includes tests or only affects documentation.
- [ ] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [ ] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-email/blob/main/CHANGELOG.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,19 +17,86 @@
...
 ## Getting Started
 
-### Python setup
+### Integrate with Prefect flows
 
-Requires an installation of Python 3.7+.
+`prefect-email` makes sending emails effortless, giving you peace of mind that your emails are being sent as expected.
 
-We recommend using a Python virtual environment manager such as pipenv, conda or virtualenv.
+First, install [prefect-email](#installation) and [save to block](#saving-credentials-to-block) to run the examples below!
```
Comment (desertaxle) [inline README.md:31]:
```suggestion
First, install [prefect-email](#installation) and [save your email credentials to a block](#saving-credentials-to-block) to run the examples below!
```

File: README.md [hunk 2]
```diff
@@ -15,19 +17,86 @@
...
+
+```python
+from prefect import flow
+from prefect.context import get_run_context
+from prefect_email import EmailServerCredentials, email_send_message
+
+@flow
+def notify_exc_by_email(exc):
+    context = get_run_context()
+    flow_run_name = context.flow_run.name
```
Comment (desertaxle) [inline README.md:77]: Would this get the name of the parent flow, or the flow that the code is currently running in?

File: README.md [hunk 3]
```diff
@@ -45,50 +114,74 @@ prefect block register -m prefect_email
 
 Note, to use the `load` method on Blocks, you must already have a block document [saved through code](https://orion-docs.prefect.io/concepts/blocks/#saving-blocks) or [saved through the UI](https://orion-docs.prefect.io/ui/blocks/).
 
-### Send an email using Gmail
+Requires an installation of Python 3.7+.
+
+We recommend using a Python virtual environment manager such as pipenv, conda or virtualenv.
+
+These tasks are designed to work with Prefect 2.0. For more information about how to use Prefect, please refer to the [Prefect documentation](https://orion-docs.prefect.io/).
```
Comment (desertaxle) [inline README.md:121]: I think the accepted nomenclature is Prefect 2
```suggestion
These tasks are designed to work with Prefect 2. For more information about how to use Prefect, please refer to the [Prefect documentation](https://orion-docs.prefect.io/).
```

File: prefect_email/credentials.py [hunk 4]
```diff
@@ -91,12 +91,40 @@ class EmailServerCredentials(Block):
...
+
+    username: Optional[str] = Field(
+        default=None,
+        description=(
+            "The username to use for authentication to the server. "
+            "Unnecessary if SMTP login is not required."
+        ),
+    )
+    password: SecretStr = Field(
+        default=SecretStr(""),
```
Comment (desertaxle) [inline prefect_email/credentials.py:104]: I think we might want a default factory here to be safe since `SecretStr` is mutable.

**Review by desertaxle — CHANGES_REQUESTED** [review]

File: prefect_email/credentials.py (hunk 4 above)
Comment (ahuang11) [reply prefect_email/credentials.py:104]: Don't think it's possible to have a default factory:
```
from pydantic import SecretStr

SecretStr()
```

```

TypeError                                 Traceback (most recent call last)
Cell In[1], line 3
      1 from pydantic import SecretStr
----> 3 SecretStr()

File ~/mambaforge/envs/email/lib/python3.10/site-packages/pydantic/types.py:888, in pydantic.types.SecretStr.__init__()

TypeError: __init__() takes exactly 2 positional arguments (1 given)
```

Comment (desertaxle) [reply prefect_email/credentials.py:104]: I think you could make one with something along the lines of `partial(SecretStr, "")`

Comment (ahuang11) [reply prefect_email/credentials.py:104]: Clever!
```
from functools import partial
from pydantic import SecretStr, Field, BaseModel

class Test(BaseModel):

    password: SecretStr = Field(default_factory=partial(SecretStr, ""))

Test()
```

**Review by desertaxle — APPROVED** [review]
LGTM!

File: README.md (hunk 2 above)
Comment (ahuang11) [reply README.md:77]: You're right. This uses the child flow. I'm simply going to remove the flow decorator here.

File: README.md [hunk 5]
```diff
@@ -15,19 +17,86 @@
...
+
+`prefect-email` can be wrapped in an `except` statement to do just that!
+
+```python
+from prefect import flow
+from prefect.context import get_run_context
+from prefect_email import EmailServerCredentials, email_send_message
 
-These tasks are designed to work with Prefect 2.0. For more information about how to use Prefect, please refer to the [Prefect documentation](https://orion-docs.prefect.io/).
+@flow
```
Comment (ahuang11) [inline README.md:74]:
```suggestion
```

**Comment by ahuang11** [comment]
Just saw your earlier comment. I removed the flow decorator and it works as expected:
image
image

**Review by desertaxle — APPROVED** [review]

**Pull request merged**